server/aws-lsp-identity/src/language-server/identityService.ts

```diff
@@ -35,13 +45,31 @@ const flows: AuthFlows = {
     [AuthorizationFlowKind.DeviceCode]: deviceCodeFlow,
     [AuthorizationFlowKind.Pkce]: authorizationCodePkceFlow,
 }
+const qPermissions = [
+    'q:StartConversation',
+    'q:SendMessage',
+    'q:GetConversation',
+    'q:ListConversations',
+    'q:UpdateConversation',
+    'q:DeleteConversation',
+    'q:PassRequest',
+    'q:StartTroubleshootingAnalysis',
+    'q:StartTroubleshootingResolutionExplanation',
+    'q:GetTroubleshootingResults',
+    'q:UpdateTroubleshootingCommandResult',
+    'q:GetIdentityMetaData',
+    'q:GenerateCodeFromCommands',
+    'q:UsePlugin',
+    'codewhisperer:GenerateRecommendations',
+]
 
 export class IdentityService {
     constructor(
         private readonly profileStore: ProfileStore,
         private readonly ssoCache: SsoCache,
         private readonly autoRefresher: SsoTokenAutoRefresher,
         private readonly handlers: SsoFlowParams['handlers'],
+        private readonly sendGetMfaCode: (params: GetMfaCodeParams) => Promise<GetMfaCodeResult>,
```

Comment:
Per the other comment, refactor the previous parameter to generically support handlers (not just SsoFlowParams, consider renaming if appropriate).  All current and future handlers should come in through that parameter.